dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java

```diff
@@ -112,4 +144,87 @@ public final Response getContainers(@Context final HttpServletRequest request,
 
         }
     }
+    
+    
+    
+    @GET
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Path("/{containerId}/content/{contentletId}")
+    public final Response containerContent(@Context final HttpServletRequest req,@Context final HttpServletResponse res,
+            @PathParam("containerId") final String containerId,
+            @PathParam("contentletId") final String contentletId) 
+                    throws DotDataException, DotSecurityException, ParseErrorException, MethodInvocationException, ResourceNotFoundException, IOException, IllegalAccessException, InstantiationException, InvocationTargetException, NoSuchMethodException {
+
+        final InitDataObject initData = webResource.init(true, req, true);
+        final User user = initData.getUser();
+        PageMode mode = PageMode.get(req);
+        Language id = WebAPILocator.getLanguageWebAPI().getLanguage(req);
+        ShortyId containerShorty = APILocator.getShortyAPI().getShorty(containerId).orElseGet(() -> {throw new ResourceNotFoundException("Can't find Container" + containerId);} );
+        
+        ShortyId contentShorty = APILocator.getShortyAPI().getShorty(contentletId).orElseGet(() -> {throw new ResourceNotFoundException("Can't find contentlet:" + contentletId);} );
+        
+        
+        
+        final Contentlet contentlet = (contentShorty.subType == ShortType.CONTENTLET) ?  APILocator.getContentletAPI().find(contentShorty.longId, user, mode.showLive) 
+                : APILocator.getContentletAPI().findContentletByIdentifier(contentShorty.longId, mode==PageMode.ANON, id.getId(), user, mode==PageMode.ANON);
+
+        if(contentlet.getContentType().baseType() == BaseContentType.WIDGET) {
+            Map<String, String> response = new HashMap<>();
+            response.put("render", WidgetResource.parseWidget(req, res, contentlet));
+            
+            return Response.ok(response).build();
+            
+            
+        }
+        
+        
+        Container container = (containerShorty.subType == ShortType.CONTAINER) 
+            ? APILocator.getContainerAPI().find(containerId, user, mode==PageMode.ANON)
+                    : (mode.showLive) 
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)